src/twosat.rs

```diff
@@ -1,23 +1,99 @@
+//! A 2-SAT Solver.
 use crate::internal_scc;
 
+/// A 2-SAT Solver.
+///
+/// For variables $x_0, x_1, \ldots, x_{N - 1}$ and clauses with from
+///
+/// \\[
+///   (x_i = f) \lor (x_j = g)
+/// \\]
+///
+/// it decides whether there is a truth assignment that satisfies all clauses.
+///
+/// # Example
+///
+/// ```
+/// #![allow(clippy::many_single_char_names)]
+///
+/// use ac_library_rs::TwoSat;
+/// use proconio::{input, marker::Bytes, source::once::OnceSource};
+///
+/// input! {
+///     from OnceSource::from(
+///         "3\n\
+///          3\n\
+///          a b\n\
+///          !b c\n\
+///          !a !a\n",
+///     ),
+///     n: usize,
+///     pqs: [(Bytes, Bytes)],
+/// }
+///
+/// let mut twosat = TwoSat::new(n);
+///
+/// for (p, q) in pqs {
+///     fn parse(s: &[u8]) -> (usize, bool) {
+///         match *s {
+///             [c] => ((c - b'a').into(), true),
+///             [b'!', c] => ((c - b'a').into(), false),
+///             _ => unreachable!(),
+///         }
+///     }
+///     let ((i, f), (j, g)) = (parse(&p), parse(&q));
+///     twosat.add_clause(i, f, j, g);
+/// }
+///
+/// assert!(twosat.satisfiable());
+/// assert_eq!(twosat.answer(), [false, true, true]);
+/// ```
 pub struct TwoSat {
     n: usize,
     scc: internal_scc::SccGraph,
     answer: Vec<bool>,
 }
 impl TwoSat {
+    /// Creates a new `TwoSat` of `n` variables and 0 clause.
```

Comment:
nit: plural (it's kind of a matter of preference, but in [the official doc](https://atcoder.github.io/ac-library/production/document_en/twosat.html) it's plural)
```suggestion
    /// Creates a new `TwoSat` of `n` variables and 0 clauses.
```

Reply:
Thank you.